libmpv/cocoa-openglcb/cocoa-openglcb.m

```diff
@@ -68,9 +68,25 @@ - (void)fillBlack
 
 - (void)drawRect
 {
-    if (self.mpvGL)
-        mpv_opengl_cb_draw(self.mpvGL, 0, self.bounds.size.width, -self.bounds.size.height);
-    else
+    if (self.mpvGL) {
+        mpv_render_param params[] = {
+            // Specify the default framebuffer (0) as target. This will
+            // render onto the entire screen. If you want to show the video
+            // in a smaller rectangle or apply fancy transformations, you'll
+            // need to render into a separate FBO and draw it manually.
+            {MPV_RENDER_PARAM_OPENGL_FBO, &(mpv_opengl_fbo){
+                .fbo = 0,
+                .w = self.bounds.size.width,
+                .h = self.bounds.size.height,
+            }},
+            // Flip rendering (needed due to flipped GL coordinate system).
+            {MPV_RENDER_PARAM_FLIP_Y, &(int){1}},
+            {MPV_RENDER_PARAM_INVALID, 0}
```

Comment:
i wouldn't use `MPV_RENDER_PARAM_INVALID` here. this is just supposed to be a terminator and just a `{0}` is probably more appropriate.